assignment1.go

```diff
@@ -0,0 +1,64 @@
+package main
+
+import (
+	"fmt"
+	"math"
+)
+
+func hashGenerator(str string) uint64 {
+	var base, hash uint64
+
+	base = 257
+	bucketSize = uint64(math.Pow(2, 64))
```

Comment:
The use of math.Pow (floating-point math) for an integer value is not idiomatic and can be error-prone.
You can use any of the syntax below
```suggestion
	bucketSize := uint64(math.MaxUint64)
```
```suggestion
		var bucketSize uint64 = math.MaxUint64
```

Also, since this value is fixed always, you can utilise a const at package level.
```suggestion
	const bucketSize uint64 = math.MaxUint64
```